### DISCUSSION TITLE
New Neighbor Search module

### DISCUSSION THREAD

**Discussion by Dgraymd** [isl-org/Open3D#3563] [Q&A] [answered]
Hello everyone,

i wonder whether i can choose between CPU and GPU to for the search algorithms, such as KNN, Radius search, and Hybrid search?

Best regards
Mohannad

**Comment by ssheorey** [comment, marked as answer]
Whether CPU or GPU is used depends on where the data is stored for constructing the `NearestNeighborSearch` object. If you want to use CPU for a specific operation, move the data to the CPU first. If you want to use both at the same time, you'll have to build two independent NNS data structures.

**Reply by germanros1987** [reply]
Please, @Dgraymd, consider making this question as answered.
